xmtp_mls/src/storage/encrypted_store/group.rs

```diff
@@ -49,9 +50,25 @@ impl StoredGroup {
             purpose: Purpose::Conversation,
         }
     }
+
+    /// Create a new [`Purpose::Sync`] group
+    pub fn new_sync_group(
+        id: ID,
+        created_at_ns: i64,
+        membership_state: GroupMembershipState,
+    ) -> Self {
+        Self {
+            id,
+            created_at_ns,
+            membership_state,
+            installations_last_checked: 0,
+            purpose: Purpose::Sync,
+        }
+    }
 }
 
 impl DbConnection<'_> {
+    /// Return  regular [`Purpose::Conversation`] groups with additional filters
     pub fn find_groups(
```

Comment:
We need to add the code change to filter out the sync groups here (or filter to Conversation groups) right?

Reply:
yep, that's already happening from my [last PR](https://github.com/xmtp/libxmtp/pull/553/files#diff-a7388e26da76ba3ef25a9251cfca5a9b66a99e7d91e3d041c6dce54e4618feeaR80) 😉